src/components/Invoice/index.jsx

```diff
@@ -1,109 +1,120 @@
-import {useState, useEffect} from "react";
+import { useState, useEffect } from "react";
 import Modal from "react-bootstrap/Modal";
 import "../../index.css";
 
-import InvoiceTable from "../InvoiceTable/index.jsx";
-import GeneralInvoiceData from "../GeneralInvoiceData/index.jsx";
+import InvoiceTable from "../InvoiceTable";
+import GeneralInvoiceData from "../GeneralInvoiceData";
+import CreateNewInvoiceButton from "../CreateNewInvoiceButton";
 
-function Invoice({invoice, shopDetails}) {
-    const {id, invoice_id, name, due, invoice_total, status, details, status_name} =
-        invoice;
+function Invoice({ invoice, shopDetails }) {
+  const {
+    id,
+    invoice_id,
+    name,
+    due,
+    invoice_total,
+    status,
+    details,
+    status_name,
+  } = invoice;
 
-    const [show, setShow] = useState(false);
-    const [idForModal, setIdForModal] = useState("");
+  const [show, setShow] = useState(false);
+  const [idForModal, setIdForModal] = useState("");
 
-    const handleClose = () => setShow(false);
-    const handleShow = (id) => {
-        setIdForModal(id);
-        setShow(true);
-    };
+  const handleClose = () => setShow(false);
+  const handleShow = (id) => {
+    setIdForModal(id);
+    setShow(true);
+  };
 
-    const [invoiceDetails, setInvoiceDetails] = useState({
-        "name": "Dummy",
-        "street_address": "Dummy",
-        "city": "Dummy",
-        "created": "Dummy",
-        "due": "Dummy",
-        'details': [{
-            'description': 'test',
-            'quantity': '1',
-            'rate': 'test',
-            'total': 'test'
-        }]
-    });
+  const [invoiceDetails, setInvoiceDetails] = useState({
+    name: "Dummy",
+    street_address: "Dummy",
+    city: "Dummy",
+    created: "Dummy",
+    due: "Dummy",
+    details: [
+      {
+        description: "test",
+        quantity: "1",
+        rate: "test",
+        total: "test",
+      },
+    ],
+  });
 
-    useEffect(() => {
-        const fetchData = async () => {
-            const response = await fetch(
-                `https://invoicing-api.dev.io-academy.uk/invoices/${invoice.id}`
-            );
-            const json = await response.json();
-            setInvoiceDetails(json.data);
-        };
-        fetchData().catch(console.error);
-    }, []);
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch(
+        `https://invoicing-api.dev.io-academy.uk/invoices/${invoice.id}`
```

Comment:
The API url is hardcoded in several places: it would be better to have it defined as a `const` so it would be easier to change if needed.